### PR TITLE
chore: upgrade useink and setup vite

### PR DESCRIPTION
**Pull request by gaboesquivel**

### CODE REVIEW

**Comment by gaboesquivel** [comment]
This complete @statictype  @DoubleOTheven, clippy and contract test have been failing since Aug 19, 2022 https://github.com/paritytech/link/commits/master. Should we remove them?

File: frontend/.eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...eact-hooks/exhaustive-deps","React Hook useMemo has missing dependencies: 'abi' and 'link.contract'. Either include them or remove the dependency array.","ArrayExpression",["157"],["158"],"React Hook useEffect has missing dependencies: 'resolve' and 'slug'. Either include them or remove the dependency array.",["159"],["160"],"React Hook useEffect has a missing dependency: 'shortenDryRun'. Either include it or remove the dependency array.",["161"],["162"],{"desc":"163","fix":"164"},{"kind":"165","justification":"166"},{"desc":"167","fix":"168"},{"kind":"165","justification":"166"},{"desc":"169","fix":"170"},{"kind":"165","justification":"166"},"Update the dependencies array to be: [abi, link.contract, resolve.result?.ok, resolve.result.value.raw]",{"range":"171","text":"172"},"directive","","Update the dependencies array to be: [resolve, slug]",{"range":"173","text":"174"},"Update the dependencies array to be: [shortenDryRun, shortenDryRun.send, values.alias, values.url]",{"range":"175","text":"176"},[1521,1538],"[abi, link.contract, resolve.result?.ok, resolve.result.value.raw]",[1786,1788],"[resolve, slug]",[921,968],"[shortenDryRun, shortenDryRun.send, values.alias, values.url]"]
```
Comment (cmichi) [inline frontend/.eslintcache:1]: Please remove this file from the PR.

File: frontend/src/hooks/useSubmitHandler.ts [hunk 2]
```diff
@@ -20,7 +20,8 @@ export const useSubmitHandler = () => {
         setSubmitting(false);
       }
 
-      if (!result?.status.isInBlock) return;
```
Comment (statictype) [inline frontend/src/hooks/useSubmitHandler.ts:23]: @gaboesquivel we shouldn't comment out broken functionality. 

@DoubleOTheven please look into why `ContractSubmittableResult` seems to incorrectly extend `SubmittableResult` in useInk.

[Here](https://github.com/polkadot-js/api/blob/c028c6958211832bae5e5dda1868466a48195415/packages/api/src/submittable/Result.ts#L37) is the expected shape.

If you undo the comments in this file, you will get build errors:

```
error TS2339: Property 'events' does not exist on type 'ContractSubmittableResult'.
error TS2339: Property 'dispatchError' does not exist on type 'ContractSubmittableResult'.
```

File: frontend/src/Resolver.tsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
-import { useAbiMessage } from "useink";
+import './App.css';
+import { useEffect, useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { Loader } from './components';
+
+import { useLinkContract } from './hooks';
+import { AbiMessage, ContractExecResult, Registry } from 'useink/core';
+import { useAbiMessage } from 'useink';
+import { hexToString } from './lib/converter';
```
Comment (statictype) [inline frontend/src/Resolver.tsx:9]: can you explain why you changed the `hexToString` import?

File: frontend/src/Resolver.tsx [hunk 4]
```diff
@@ -48,16 +49,20 @@ const Resolver: React.FC = () => {
...
   return (
-    <div className="App h-screen flex flex-col justify-center">
-      {resolve?.isSubmitting && <Loader message={resolvedUrl ? `Redirecting to ${resolvedUrl}` : ""} />}
+    <div className="flex flex-col justify-center h-screen App">
+      {resolve?.isSubmitting && <Loader message={resolvedUrl ? `Redirecting to ${resolvedUrl}` : ''} />}
       {!resolve?.isSubmitting && resolve?.result?.ok && !resolvedUrl && (
         <div className="text-center">
-          <h1 className="text-2xl font-bold text-center">URL not found for <span className="bg-white/5 rounded-md p-2">{slug}</span></h1>
-          <p className="mt-6">Go back to <a href="/">shortener</a>.</p>
+          <h1 className="text-2xl font-bold text-center">
```
Comment (statictype) [inline frontend/src/Resolver.tsx:60]: there are several empty new lines

File: frontend/tsconfig.json [hunk 5]
```diff
@@ -1,29 +1,29 @@
...
-    "allowSyntheticDefaultImports": true,
-    "strict": true,
-    "forceConsistentCasingInFileNames": true,
-    "noFallthroughCasesInSwitch": true,
-    "module": "esnext",
+    "target": "ES2020",
+    "useDefineForClassFields": true,
+    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    "module": "ESNext",
+    "skipLibCheck": true,
```
Comment (statictype) [inline frontend/tsconfig.json:7]: please change `skipLibCheck` back to false

File: frontend/vite.config.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import svgr from 'vite-plugin-svgr';
+import nodePolyfills from 'vite-plugin-node-stdlib-browser';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react(), tsConfigPaths(), eslintPlugin(), svgr(), nodePolyfills()],
+
+  server: { port: 8081 },
+  build: {
+    minify: false, // solves vite [WARNING] Expected ";" [css-syntax-error]
```
Comment (statictype) [inline frontend/vite.config.ts:17]: disabling minification is not a good idea in any case. the bundle size got larger.

**Review by statictype — CHANGES_REQUESTED** [review]

**Comment by statictype** [comment]
> This complete @statictype @DoubleOTheven, clippy and contract test have been failing since Aug 19, 2022 https://github.com/paritytech/link/commits/master. Should we remove them?

Sometimes those tests fail due to CI problems, not actual broken functionality. This is probably the reason, we haven't touched the contract code. That part would really need some love, it's still on ink! v3.  wdyt @cmichi?

File: frontend/src/hooks/useSubmitHandler.ts (hunk 2 above)
Comment (gaboesquivel) [reply frontend/src/hooks/useSubmitHandler.ts:23]: good catch, I didn't realized that was missing

File: frontend/src/Resolver.tsx (hunk 3 above)
Comment (gaboesquivel) [reply frontend/src/Resolver.tsx:9]: it seems the new version of useink doesnt export that function

File: frontend/src/Resolver.tsx (hunk 4 above)
Comment (gaboesquivel) [reply frontend/src/Resolver.tsx:60]: its the tailwind formatting, I had autoformat on save turned on

File: frontend/tsconfig.json (hunk 5 above)
Comment (gaboesquivel) [reply frontend/tsconfig.json:7]: @DoubleOTheven when I disable skipLibCheck compiler is not able to find many dependencies of `useink`.  
Perhaps they should be defined as dependencies or specified as peer dependencies?

**Pull request closed without merging**